deepchecks/base/check_context.py

```diff
@@ -0,0 +1,181 @@
+import abc
+from typing import Callable, Union, Mapping, Sequence
+
+import pandas as pd
+
+from deepchecks import BaseCheck, CheckResult, Dataset
+from deepchecks.utils.validation import validate_model
+from errors import DatasetValidationError, ModelValidationError
+from utils.metrics import ModelType, task_type_check, get_scorers_or_default, get_single_scorer_or_default
+
+
+class CheckRunContext:
+
+    def __init__(self,
+                 train=None,
+                 test=None,
+                 model=None,
+                 feature_importance_mode: str = 'model',
+                 scorers: Mapping[str, Union[str, Callable]] = None,
+                 non_avg_scorers: Mapping[str, Union[str, Callable]] = None
+                 ):
+        # Validations
+        if train and isinstance(train, pd.DataFrame):
+            train = Dataset(train, cat_features=[])
+        if test and isinstance(test, pd.DataFrame):
+            test = Dataset(test, cat_features=[])
+        # If both datasets, validate they fit each other
+        if train and test:
+            if not Dataset.datasets_share_label(train, test):
+                raise DatasetValidationError('train and test requires to have and to share the same label')
+            if not Dataset.datasets_share_features(train, test):
+                raise DatasetValidationError('train and test requires to share the same features columns')
+            if not Dataset.datasets_share_categorical_features(train, test):
+                raise DatasetValidationError(
+                    'train and test datasets should share '
+                    'the same categorical features. Possible reason is that some columns were'
+                    'inferred incorrectly as categorical features. To fix this, manually edit the '
+                    'categorical features using Dataset(cat_features=<list_of_features>'
+                )
+            if not Dataset.datasets_share_index(train, test):
+                raise DatasetValidationError('train and test requires to share the same index column')
+
+        if train and model:
+            validate_model(train, model)
+
+        self._train = train
+        self._test = test
+        self._model = model
+        self._feature_importance_mode = feature_importance_mode
+        self._features_importance = None
+        self._task_type = None
+        self._user_scorers = scorers
```

Comment:
```suggestion
        self._user_avg_scorers = scorers
```

Reply:
why? if avg is the default behavior

Reply:
does the user know that when setting these parameters? I thought that in this case stating stuff explicitly is better.

Reply:
I think he knows. this is also the default behavior of skelarn